### PR TITLE
Add git commit id to llvm.ident metadata

### PR DESCRIPTION
**Pull request by whitneywhtsang**
Addressing review comment: https://github.com/onnx/onnx-mlir/pull/1211#discussion_r818247185

Signed-off-by: Whitney Tsang <whitneyt@ca.ibm.com>

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -149,3 +149,4 @@ add_subdirectory(include)
 add_subdirectory(src)
 add_subdirectory(docs)
 add_subdirectory(test)
+add_subdirectory(Version)
```
Comment (etiotto) [inline CMakeLists.txt:152]: [nit]: use lower case so Version -> version

File: src/Compiler/CompilerUtils.cpp [hunk 2]
```diff
@@ -34,13 +34,21 @@
...
 #define DEBUG_TYPE "compiler_utils"
 
 using namespace std;
 using namespace mlir;
 using namespace onnx_mlir;
 
 const string OnnxMlirEnvOptionName = "ONNX_MLIR_FLAGS";
+#if defined(ONNX_MLIR_REPOSITORY) && defined(ONNX_MLIR_REVISION)
+const string OnnxMlirVersion =
+    "onnx-mlir version 1.0.0 (" ONNX_MLIR_REPOSITORY " " ONNX_MLIR_REVISION ")";
```
Comment (etiotto) [inline src/Compiler/CompilerUtils.cpp:48]: Is it possible to also get the commit for LLVM itself ?  Ideally we want to know the LLVM commit used to build onnx-mlir, as well as the commit for onnx-mlir.

**Comment by etiotto** [comment]
@jenkins-droid test this please

**Comment by etiotto** [comment]
@jenkins-droid test this please

File: version/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,23 @@
+set(llvm_source_dir ${LLVM_DIR})
```
Comment (whitneywhtsang) [inline version/CMakeLists.txt:1]: Do I need to put copyright or some sort of license?

I see from other CMakeLists.txt files, they have `SPDX-License-Identifier: Apache-2.0`, should I use the same one here?

File: src/Compiler/CompilerUtils.cpp [hunk 4]
```diff
@@ -34,13 +34,23 @@
...
 #define DEBUG_TYPE "compiler_utils"
 
 using namespace std;
 using namespace mlir;
 using namespace onnx_mlir;
 
 const string OnnxMlirEnvOptionName = "ONNX_MLIR_FLAGS";
+#if defined(ONNX_MLIR_REPOSITORY) && defined(ONNX_MLIR_REVISION) &&            \
+    defined(LLVM_REPOSITORY) && defined(LLVM_REVISION)
+const string OnnxMlirVersion =
```
Comment (etiotto) [inline src/Compiler/CompilerUtils.cpp:48]: Suggest to make it a static declaration.

File: src/Compiler/CompilerUtils.cpp [hunk 5]
```diff
@@ -535,7 +545,7 @@ static void genLLVMBitcode(const mlir::OwningOpRef<ModuleOp> &module,
   // Emit the onnx-mlir version as llvm.ident metadata.
   llvm::NamedMDNode *identMetadata =
       llvmModule->getOrInsertNamedMetadata("llvm.ident");
-  std::string version = "onnx-mlir version 1.0.0";
+  std::string version = OnnxMlirVersion;
```
Comment (etiotto) [inline src/Compiler/CompilerUtils.cpp:548]: Don't think a new variable is necessary.

**Comment by etiotto** [comment]
@jenkins-droid test this please

**Review by etiotto — APPROVED** [review]
LGTM thanks.

**Comment by AlexandreEichenberger** [comment]
Please squash and merge, and edit the message (after clicking "squash and merge", before clicking "confirm") to a single liner explaining the change.

@whitneywhtsang Thanks

**Pull request merged**